File: backend_api/routes/order.js
const express=require("express");
const orderRouter=express.Router();
const Order = require("../models/order");
// post route for creating Router
orderRouter.post("/api/orders",async(req,res) =>{
    try{
        const {
            fullName,
            email,
            state,
            city,
            locality,
            productName,
            productPrice,
            quantity,
            category,
            image,
            buyerId,
            vendorId,
            
        } =req.body;
        createdAt=new Date().getMilliseconds();
        const order=new Order({
            fullName,
            email,
            state,
            city,
            locality,
            productName,
            productPrice,
            quantity,
            category,
            image,
            buyerId,
            vendorId,
            createdAt
        });
        await order.save();
        return res.status(201).json(order);
    }
    catch(error){
        res.status(500).json({error});
        
    }
});
// get route for fetching items based on buyerid
orderRouter.get("/api/orders/:buyerId",async(req,res) => {
    try{
        const{buyerId} =req.params;
        const orders=await Order.find({buyerId});
        // if no orders found return 404 status
        if(orders.length ==0){
            return res.status(404).json({msg:"NO orders found this buyer"});
        }
        return res.status(200).json(orders);

    }
    catch(e){
        res.status(500).json({error:e.message});
    }


});
// delete route to delete item from order
orderRouter.delete("/api/orders/:id",async(req,res) => {
    try{
        // extract item based on id;
        const {id} =req.params;
        // find item and delete from the data based on id
        const deletedorder=await Order.findByIdAndDelete(_id);
        // check if order was found and deleted
        if(!deletedorder){
            // if no order found with provided id return 404
            return res.status(404).json({msg:"Order not found"});
        }
        else{
            //if order successfully deleted provide status code 200
            return res.status(200).json({msg:"order was deleted sucessfully"});
        }


    }
    catch(e){
        return res.status(500).json({error:"an internal error has occured"});

    }
})
module.exports=orderRouter;

File: lib/models/order.dart
// ignore_for_file: public_member_api_docs, sort_constructors_first
import 'dart:convert';

class Order {
  final String id;
  final String fullName;
  final String email;
  final String state;
  final String city;
  final String locality;
  final String productName;
  final int productPrice;
  final int quantity;
  final String category;
  final String image;
  final String buyerId;
  final String vendorId;
  final bool processing;
  final bool delivered;
 

  Order({required this.id, required this.fullName, required this.email, required this.state, required this.city, required this.locality, required this.productName, required this.productPrice, required this.quantity, required this.category, required this.image, required this.buyerId, required this.vendorId, required this.processing, required this.delivered});

  Map<String, dynamic> toMap() {
    return <String, dynamic>{
      '_id': id,
      'fullName': fullName,
      'email': email,
      'state': state,
      'city': city,
      'locality': locality,
      'productName': productName,
      'productPrice': productPrice,
      'quantity': quantity,
      'category': category,
      'image': image,
      'buyerId': buyerId,
      'vendorId': vendorId,
      'processing': processing,
      'delivered': delivered,
    };
  }
    String toJson() => json.encode(toMap());

  factory Order.fromJson(Map<String, dynamic> map) {
    return Order(
      id: (map["_id"] ?? '') as String,
      fullName: (map["fullName"] ?? '') as String,
      email: (map["email"] ?? '') as String,
      state: (map["state"] ?? '') as String,
      city: (map["city"] ?? '') as String,
      locality: (map["locality"] ?? '') as String,
      productName: (map["productName"] ?? '') as String,
      productPrice: (map["productPrice"] ?? 0) as int,
      quantity: (map["quantity"] ?? 0) as int,
      category: (map["category"] ?? '') as String,
      image: (map["image"] ?? '') as String,
      buyerId: (map["buyerId"] ?? '') as String,
      vendorId: (map["vendorId"] ?? '') as String,
      processing: (map["processing"] ?? false) as bool,
      delivered: (map["delivered"] ?? false) as bool,
      
    );
  }



  // factory Order.fromJson(String source) => Order.fromMap(json.decode(source) as Map<String, dynamic>);
}
